#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, str_t

gen = ParameterGenerator()

#map params
gen.add("map_z_min", double_t, 0, "The depth (minimum z) of the map in meters.", 0.0)
gen.add("map_z_max", double_t, 0, "The height (maximum z) of the map in meters.", 2.05)

# robot 3d footprint
gen.add("footprint_3d_padding", double_t, 0, "How much to pad (increase the size of) the 3D footprint, in meters.", 0.01)

exit(gen.generate("costmap_3d", "costmap_3d", "Costmap3D"))
